### PR TITLE
Made mesh upload to gpu optional

### PR DESCRIPTION
**Pull request by Sudar-shan**
#85 
Exposed uploadMeshOnLoad to GltfAsset , so it can be used to manipulate mesh after upload.

### CODE REVIEW

**Comment by atteneder** [comment]
Hi Sudarsha,

Thanks for the PR!

Before review, could you explain your use case a bit more?

As far as I understand you need the mesh data in-memory in order to create collider meshes for physics use. Do you simply traverse the GameObject's hierarchy and add MeshCollider components to all GameObjects with MeshRenderers attached?

Since I imagine this is a commonly used operation, I'm thinking about integrating it a bit better in the future.

Thanks!

**Comment by Sudar-shan** [comment]
> As far as I understand you need the mesh data in-memory in order to create collider meshes for physics use. Do you simply traverse the GameObject's hierarchy and add MeshCollider components to all GameObjects with MeshRenderers attached?
> 
> Since I imagine this is a commonly used operation, I'm thinking about integrating it a bit better in the future.
> 
> Thanks!

Yes andreas,
we have a list of GameObjects with MeshRenderers and SkinnedMeshRenderers.
I create colliders(Mesh Colliders) inside a loop.

File: Runtime/Scripts/GltFast.cs [hunk 1]
```diff
@@ -196,7 +198,7 @@ enum ImageFormat {
             monoBehaviour.StartCoroutine(LoadRoutine(url,gltfBinary));
         }
 
-        IEnumerator LoadRoutine( string url, bool gltfBinary ) {
```
Comment (atteneder) [inline Runtime/Scripts/GltFast.cs:199]: space only changes mess up the history! pleasee undo

**Review by atteneder — COMMENTED** [review]
Thanks for the contribution!

There's a non-change in `Tests/Textures/colorgrid.png`. Any idea where the change comes from? maybe some git lfs trouble. Please remove change.

**Comment by atteneder** [comment]
Again, thanks for this contribution. I finally took the time to think it through.

glTFast at the moment discards imported meshes from main memory after uploading them to the GPU.
In some use cases that's not desirable (most prominent: create a collider mesh). Question is on which level this should be optional:

- Globally (via scripting define)
  - pro: easy and fast (no runtime impact at all)
  - con: mixed usage comes at a memory cost (all meshes stick in memory, regardless if they are used)
- Per asset (via parameter...the way this PR works)
  - pro: a bit more flexible
  - con: mild overhead
- Per mesh/primitive (user defined via IInstantiator injection)
  - pro: most flexible solution. Allows to treat collider meshes different than render meshes with custom logic
  - con: most effort to implement

My estimation:

Turning mesh discarding on globally should be done in any way.
I'm having a hard time imagining a use-case where you need ALL meshes of some asset and NONE of others. Per asset parameter seems not so useful to me.
Per mesh/primitive is a powerful final solution (for the mentioned case).

For now I'd say let's enable mesh discarding by default with the option of disabling it globally. At a later point implement it per mesh/primitive.

Let me know your thoughts.

Thanks!

**Comment by Sudar-shan** [comment]
Hey Andreas,
sry was out a bit . so late reply.
Looks good. I will make necessary changes. Thanks.

**Comment by Sudar-shan** [comment]
Hi Andreas,
sorry was out a bit. so late reply.
Noted your comments on code style.
Global is ok with me.
Thanks.

**Pull request closed without merging**